azplugins/flow.py

```diff
@@ -746,13 +767,51 @@ def density(self):
             return np.zeros(self.bins)
 
     @property
-    def velocity(self):
-        r"""The current average velocity profile."""
+    def mass_density(self):
+        r"""The current mass-averaged density profile."""
+        if hoomd.comm.get_rank() != 0:
+            hoomd.context.msg.error('Flow profile only defined on root rank.\n')
+            raise RuntimeError('Flow profile only defined on root rank')
+
+        if self.samples > 0:
+            return np.divide(self._bin_mass, self._counts, out=np.zeros(self.bins), where=self._counts > 0)
+        else:
+            return np.zeros(self.bins)
+
+    @property
+    def number_velocity(self):
+        r"""The current number-averaged velocity profile ."""
+        if hoomd.comm.get_rank() != 0:
+            hoomd.context.msg.error('Flow profile only defined on root rank.\n')
+            raise RuntimeError('Flow profile only defined on root rank')
+
+        result = np.zeros_like(self._number_velocity)
+        if self.samples > 0:
+            for dim in range(3):
+               result[dim] = self._number_velocity[dim]/self.samples
+        return result
+
+    @property
+    def mass_velocity(self):
+        r"""The current mass-averaged velocity profile ."""
+        if hoomd.comm.get_rank() != 0:
+            hoomd.context.msg.error('Flow profile only defined on root rank.\n')
+            raise RuntimeError('Flow profile only defined on root rank')
+
+        result = np.zeros_like(self._mass_velocity)
+        if self.samples > 0:
+            for dim in range(3):
+               result[dim] = np.divide(self._mass_velocity[dim], self._counts, out=np.zeros(self.bins), where=self._counts > 0)
+        return result
+
+    @property
+    def kT(self):
+        r"""The current average temperature profile."""
         if hoomd.comm.get_rank() != 0:
             hoomd.context.msg.error('Flow profile only defined on root rank.\n')
             raise RuntimeError('Flow profile only defined on root rank')
 
         if self.samples > 0:
-            return np.divide(self._velocity, self._counts, out=np.zeros(self.bins), where=self._counts > 0)
+            return np.divide(self._vsqm, (3*(self._counts-1)), out=np.zeros(self.bins), where=(3*(self._counts-1)) > 0)
```

Comment:
I think if we correct code above, this will also look like the velocity profile code, just divide the internal kT by samples.